open-codegen/opengen/templates/python/python_bindings.rs

```diff
@@ -0,0 +1,144 @@
+
+        // ----------------------------------------------------
+        // Set initial value
+        // ----------------------------------------------------
+        if let Some(u0) = initial_guess {
+            if u0.len() != {{meta.optimizer_name|upper}}_NUM_DECISION_VARIABLES {
+                println!(
+                    "1600 -> Initial guess has incompatible dimensions: {} != {}",
+                    u0.len(),
+                    {{meta.optimizer_name|upper}}_NUM_DECISION_VARIABLES
+                );
+                return Ok(None);
+            }
+            u.copy_from_slice(&u0);
+        }
+
+        // ----------------------------------------------------
+        // Check lagrange multipliers
+        // ----------------------------------------------------
+        if let Some(y0) = &initial_lagrange_multipliers {
+            if y0.len() != {{meta.optimizer_name|upper}}_N1 {
+                println!(
+                    "1700 -> wrong dimension of Langrange multipliers: {} != {}",
+                    y0.len(),
+                    {{meta.optimizer_name|upper}}_N1
+                );
+                return Ok(None);
+            }
+        }
+
+        // ----------------------------------------------------
+        // Check parameter
+        // ----------------------------------------------------
+        if p.len() != {{meta.optimizer_name|upper}}_NUM_PARAMETERS {
+            println!(
+                "3003 -> wrong number of parameters: {} != {}",
+                p.len(),
+                {{meta.optimizer_name|upper}}_NUM_PARAMETERS
+            );
+            return Ok(None);
+        }
+
+        // ----------------------------------------------------
+        // Run solver
+        // ----------------------------------------------------
+        let solver_status = solve(
+            &p,
+            &mut self.cache,
+            &mut u,
+            &initial_lagrange_multipliers,
+            &initial_penalty,
+        );
+
+        match solver_status {
+            Ok(status) => Ok(Some(OptimizerSolution {
+                exit_status: format!("{:?}", status.exit_status()),
+                num_outer_iterations: status.num_outer_iterations(),
+                num_inner_iterations: status.num_inner_iterations(),
+                last_problem_norm_fpr: status.last_problem_norm_fpr(),
+                f1_infeasibility: status.delta_y_norm_over_c(),
+                f2_norm: status.f2_norm(),
+                penalty: status.penalty(),
+                lagrange_multipliers: status.lagrange_multipliers().clone().unwrap_or_default(),
+                solve_time_ms: (status.solve_time().as_nanos() as f64) / 1e6,
+                solution: u.to_vec(),
+                cost: status.cost(),
+            })),
+            Err(_) => {
+                println!("2000 -> Problem solution failed (solver error)");
+                Ok(None)
```

Comment:
One thing I'd like to do would be to return the error value some how instead of just printing to stdout. But we don't need to block on this. Lets do it in a follow up PR

Reply:
I agree. Let's do that right afterwards.